src/osd/OSD.cc

```diff
@@ -5517,6 +5520,11 @@ void OSD::flush_pg_stats()
   osd_lock.Lock();
 }
 
+void OSD::send_beacon()
+{
+  dout(20) << __func__ << dendl;
+  monc->send_mon_message(new MOSDBeacon());
```

Comment:
I think we'll want to throttle this a bit; the failure and full update reports go out pretty quickly, but these should only be every several minutes to minimize mon load.